### ISSUE TITLE
OrientVertex.removeProperty breaks automatic detach of null properties

### ISSUE THREAD

**Issue by vitorenesduarte** [orientechnologies/orientdb#3894]
Hi,
__v2.0.6__

[This](https://gist.github.com/vitorenesduarte/834d81feef7375a2c971)' piece of code throws this exception:
```bash
Exception in thread "main" com.orientechnologies.orient.core.exception.ODatabaseException: Database instance is not set in current thread. Assure to set it with: ODatabaseRecordThreadLocal.INSTANCE.set(db);
	at com.orientechnologies.orient.core.db.ODatabaseRecordThreadLocal.get(ODatabaseRecordThreadLocal.java:54)
	at com.orientechnologies.orient.core.record.ORecordAbstract.getDatabase(ORecordAbstract.java:240)
	at com.orientechnologies.orient.core.record.impl.ODocument.getGlobalPropertyById(ODocument.java:1938)
	at com.orientechnologies.orient.core.record.impl.ODocumentInternal.getGlobalPropertyById(ODocumentInternal.java:55)
	at com.orientechnologies.orient.core.serialization.serializer.record.binary.ORecordSerializerBinaryV0.getGlobalProperty(ORecordSerializerBinaryV0.java:275)
	at com.orientechnologies.orient.core.serialization.serializer.record.binary.ORecordSerializerBinaryV0.deserializePartial(ORecordSerializerBinaryV0.java:127)
	at com.orientechnologies.orient.core.serialization.serializer.record.binary.ORecordSerializerBinary.fromStream(ORecordSerializerBinary.java:72)
	at com.orientechnologies.orient.core.record.impl.ODocument.deserializeFields(ODocument.java:1755)
	at com.orientechnologies.orient.core.record.impl.ODocument.checkForFields(ODocument.java:2187)
	at com.orientechnologies.orient.core.record.impl.ODocument.rawField(ODocument.java:834)
	at com.orientechnologies.orient.core.record.impl.ODocument.field(ODocument.java:854)
	at com.tinkerpop.blueprints.impls.orient.OrientElement.getProperty(OrientElement.java:251)
	at Detach.main(Detach.java:97)
```

It seems that __removeProperty__ is breaking something.
If we comment from line A to line B, everything works normally.

**Comment by lvca** [comment]
Accessing to a field after closing the database could raise this exception if hasn't been loaded. Call `.detach()` while the graph is open.

    phoebeDetached.detach();

**Comment by vitorenesduarte** [comment]
@lvca it's __removeProperty__ that causes this exception.
Since OrientDB provides automatic detach, I've removed all __detach__ calls from my application, and it's working great unless, removeProperty is called.

**Comment by vitorenesduarte** [comment]
And it's only for properties that are __null__. This is not an issue?

**Comment by vitorenesduarte** [comment]
Just tried with SNAPSHOT, thanks!

**Comment by lvca** [comment]
The problem is because the property is not found between properties and it try to deserialize from the stream, but for this task the database is needed. That's the reason of the issue. I've added an additional check to avoid this problem in 2.0.7-SNAPSHOT and 2.1-SNAPSHOT.
